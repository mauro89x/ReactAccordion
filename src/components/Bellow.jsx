import React, { Component } from 'react';

const Bellow = ({ title, text, handleChange, index, currentIndex }) => (
    <div style={bellowContainer}>
        <a href="#" style={titleStyle} onClick={() => handleChange(index)}>{title}</a>
        {
            currentIndex === index &&
                <p style={bellowText}>{text}</p>
        }
    </div>
);

const titleStyle = {
    cursor: "pointer",
};

const bellowContainer = {
    border: "1px solid rgba(0,0,0,.125)",
    padding: 10,
    backgroundColor: "rgba(0,0,0,.03)",
    marginTop: 4
};

const bellowText = {
    backgroundColor: "white",
    padding: 20
};

export default Bellow;
