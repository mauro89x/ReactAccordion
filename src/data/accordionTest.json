{   
    "accordion": [
        { "title": "First Bellow", "text": "Hey test 1" },
        { "title": "Second Bellow", "text": "Hey test 2" },
        { "title": "Third Bellow", "text": "Hey test 3" },
        { "title": "Fourth Bellow", "text": "Hey test 4" }
    ]
}
