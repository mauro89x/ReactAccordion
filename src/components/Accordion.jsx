import React, { Component } from 'react';
import Bellow from './Bellow';

class Accordion extends Component {
    constructor(props) {
        super(props);
        this.state = {
            currentIndex: -1,
        }
        this.handleChange = this.handleChange.bind(this);
    }

    handleChange(i) {
        this.setState({
            currentIndex: i,
        });
    }

    render() {
        const { data: { accordion } } = this.props;
        return accordion.map((elem, i) => (
            <Bellow key={i}
                title={elem.title}
                text={elem.text}
                index={i}
                currentIndex={this.state.currentIndex}
                handleChange={this.handleChange}
            />
        ));
    }
}

export default Accordion;
