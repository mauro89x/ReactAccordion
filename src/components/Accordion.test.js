import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer'
import Accordion from './Accordion';
import data from '../data/accordionTest';
import Bellow from './Bellow';

describe('Accordion component', () => {
    var component, tree;

    beforeEach(() => {
        component = renderer.create(<Accordion data={data} />);
        tree = component.toJSON();
    });

    it('SHOULD match the snapshot', () => {
        expect(tree).toMatchSnapshot();
    });
});

describe('Testing Accordion with 4 Bellows', () => {
    var component, componentInstance, bellows;

    beforeEach(() => {
        component = renderer.create(<Accordion data={data} />);
        componentInstance = component.root;
        bellows = componentInstance.findAllByType(Bellow);
    });

    it('SHOULD render four bellows', () => {
        expect(bellows.length).toEqual(4);
    });

    it('Bellows should have the correct title title = \'First Bellow, Second Below ...\'', () => {
        expect(bellows[0].props.title).toEqual('First Bellow');
        expect(bellows[1].props.title).toEqual('Second Bellow');
        expect(bellows[2].props.title).toEqual('Third Bellow');
        expect(bellows[3].props.title).toEqual('Fourth Bellow');
    });

    it('Bellows should have the correct text', () => {
        expect(bellows[0].props.text).toEqual('Hey test 1');
        expect(bellows[1].props.text).toEqual('Hey test 2');
        expect(bellows[2].props.text).toEqual('Hey test 3');
        expect(bellows[3].props.text).toEqual('Hey test 4');
    });
})
