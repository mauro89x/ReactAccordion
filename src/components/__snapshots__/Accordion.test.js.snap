// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion component SHOULD match the snapshot 1`] = `
Array [
  <div
    style={
      Object {
        "backgroundColor": "rgba(0,0,0,.03)",
        "border": "1px solid rgba(0,0,0,.125)",
        "marginTop": 4,
        "padding": 10,
      }
    }
  >
    <a
      href="#"
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
        }
      }
    >
      First Bellow
    </a>
  </div>,
  <div
    style={
      Object {
        "backgroundColor": "rgba(0,0,0,.03)",
        "border": "1px solid rgba(0,0,0,.125)",
        "marginTop": 4,
        "padding": 10,
      }
    }
  >
    <a
      href="#"
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
        }
      }
    >
      Second Bellow
    </a>
  </div>,
  <div
    style={
      Object {
        "backgroundColor": "rgba(0,0,0,.03)",
        "border": "1px solid rgba(0,0,0,.125)",
        "marginTop": 4,
        "padding": 10,
      }
    }
  >
    <a
      href="#"
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
        }
      }
    >
      Third Bellow
    </a>
  </div>,
  <div
    style={
      Object {
        "backgroundColor": "rgba(0,0,0,.03)",
        "border": "1px solid rgba(0,0,0,.125)",
        "marginTop": 4,
        "padding": 10,
      }
    }
  >
    <a
      href="#"
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
        }
      }
    >
      Fourth Bellow
    </a>
  </div>,
]
`;
